### PR TITLE
test(badge): typescript refactor

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
- Rename the `badge.cy.js` to `badge.cy.tsx` file in `./cypress/components/badge/` folder.
- Refactor tests to Typescript.  

### Current behaviour

Currently Badge component is in JS not in TS.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the`badge.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other *.cy.tsx files have regressed

### CODE REVIEW

File: cypress/components/badge/badge.cy.tsx [hunk 1]
```diff
@@ -1,28 +1,26 @@
 import React from "react";
 import Badge from "../../../src/components/badge";
 import Button from "../../../src/components/button";
+import Box from "../../../src/components/box";
 import CypressMountWithProviders from "../../support/component-helper/cypress-mount";
 import { badge, badgeCounter, badgeCrossIcon } from "../../locators/badge";
 import { CHARACTERS } from "../../support/component-helper/constants";
 
 const BadgeComponent = ({ ...props }) => {
```
Comment (stephenogorman) [inline cypress/components/badge/badge.cy.tsx:9]: Can you move this story to ./src/components/badge/badge-test.stories.tsx?

**Review by stephenogorman — CHANGES_REQUESTED** [review]

File: cypress/components/badge/badge.cy.tsx [hunk 2]
```diff
@@ -58,7 +56,7 @@ context("Testing Badge component", () => {
       CypressMountWithProviders(
         <BadgeComponent onClick={() => {}} counter="99" />
       );
-
+      // eslint-disable-next-line jest/valid-expect-in-promise
```
Comment (DlgSHi) [inline cypress/components/badge/badge.cy.tsx:59]: why do we need this ?

File: cypress/components/badge/badge.cy.tsx [hunk 3]
```diff
@@ -70,7 +68,7 @@ context("Testing Badge component", () => {
 
     it("badge should not display cross icon when hovered over with no onClick function passed to component", () => {
       CypressMountWithProviders(<BadgeComponent counter="99" />);
-
+      // eslint-disable-next-line jest/valid-expect-in-promise
```
Comment (DlgSHi) [inline cypress/components/badge/badge.cy.tsx:71]: same here?

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: cypress/components/badge/badge.cy.tsx (hunk 2 above)
Comment (divyajindel) [reply cypress/components/badge/badge.cy.tsx:45]: Screenshot 2023-05-12 at 11 45 54


I was getting these errors while doing 'npm run commit' so Robin suggested me to add // eslint-disable-next-line jest/valid-expect-in-promise.

**Review by DlgSHi — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**